### PR TITLE
Flyin - VSCode default directory

### PR DESCRIPTION
**Pull request by troychiu**
## Tracking issue
_https://github.com/flyteorg/flyte/issues/4284_

## Why are the changes needed?
Previously, when users accessed the VSCode server by navigating to `http://localhost:8080` in their web browser, they were not directed to a specific directory. This could be confusing, especially for those who are not familiar with VSCode, as they might not see their code files immediately.

## What changes were proposed in this pull request?
The proposed change sets a default directory for the VSCode server. Now, when users connect to the server without setting the folder argument such as `http://localhost:8080`, VSCode server will automatically open the directory where the task function's source files are located, which is `http://localhost:8080/?folder=/root`.
截圖 2023-12-14 上午12 07 45

## Check all the applicable boxes 

- [X] I updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] All commits are signed-off.

### CODE REVIEW

File: flytekit/models/literals.py [hunk 1]
```diff
@@ -8,11 +8,10 @@
 from flytekit.exceptions import user as _user_exceptions
 from flytekit.models import common as _common
 from flytekit.models.core import types as _core_types
-from flytekit.models.types import Error
```
Comment (ByronHsu) [inline flytekit/models/literals.py:11]: nit: revert the change?

**Review by ByronHsu — COMMENTED** [review]
Thanks for the quick improvement!

**Review by pingsutw — APPROVED** [review]
